src/catalog/src/durable/transaction.rs

```diff
@@ -1235,6 +1231,15 @@ impl<'a> Transaction<'a> {
         Ok(())
     }
 
+    /// Get the value of a persisted config.
+    pub fn get_config(&mut self, key: String) -> Result<Option<u64>, CatalogError> {
```

Comment:
Why not `pub fn get_config(&mut self, key: String) -> Option<u64>`? If I'm reading correctly this can never return an `Err`.

Reply:
I wrapped it in a result just for symmetry with `pub fn set_config(...)`, I'll change it to what you suggest though